include/exadg/fluid_structure_interaction/driver_precice_solid.h

```diff
@@ -0,0 +1,329 @@
+      structure_operator = std::make_shared<Structure::Operator<dim, Number>>(
+        this->application->get_grid_structure(),
+        this->application->get_boundary_descriptor_structure(),
+        this->application->get_field_functions_structure(),
+        this->application->get_material_descriptor_structure(),
+        this->application->get_parameters_structure(),
+        "elasticity",
+        this->mpi_comm);
+
+      // initialize matrix_free
+      structure_matrix_free_data = std::make_shared<MatrixFreeData<dim, Number>>();
+      structure_matrix_free_data->append(structure_operator);
+
+      structure_matrix_free = std::make_shared<dealii::MatrixFree<dim, Number>>();
+      structure_matrix_free->reinit(*this->application->get_grid_structure()->mapping,
+                                    structure_matrix_free_data->get_dof_handler_vector(),
+                                    structure_matrix_free_data->get_constraint_vector(),
+                                    structure_matrix_free_data->get_quadrature_vector(),
+                                    structure_matrix_free_data->data);
+
+      structure_operator->setup(structure_matrix_free, structure_matrix_free_data);
+
+      // initialize postprocessor
+      structure_postprocessor = this->application->create_postprocessor_structure();
+      structure_postprocessor->setup(structure_operator->get_dof_handler(),
+                                     *this->application->get_grid_structure()->mapping);
+
+      this->timer_tree.insert({"FSI", "Setup", "Structure"}, timer_local.wall_time());
+    }
+
+    /**************************************** STRUCTURE *****************************************/
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    this->precice =
+      std::make_shared<ExaDG::preCICE::Adapter<dim, dim, VectorType>>(this->precice_parameters,
+                                                                      this->mpi_comm);
+
+    this->precice->add_write_surface(
+      this->application->get_boundary_descriptor_structure()->neumann_cached_bc.begin()->first,
+      this->precice_parameters.write_mesh_name,
+      {this->precice_parameters.displacement_data_name,
+       this->precice_parameters.velocity_data_name},
+      this->precice_parameters.write_data_type,
+      structure_matrix_free,
+      structure_operator->get_dof_index(),
+      numbers::invalid_unsigned_int);
+
+    {
+      std::vector<unsigned int> quad_indices;
+      quad_indices.emplace_back(structure_operator->get_quad_index());
+
+      // VectorType stress_fluid;
+      auto exadg_terminal = std::make_shared<ExaDG::preCICE::InterfaceCoupling<dim, dim, Number>>();
+      auto quadrature_point_locations = exadg_terminal->setup(
+        structure_matrix_free,
+        structure_operator->get_dof_index(),
+        quad_indices,
+        this->application->get_boundary_descriptor_structure()->neumann_cached_bc);
```

Comment:
The same as in comment https://github.com/exadg/exadg/pull/148/files#r803717139

Reply:
@DavidSCN did you resolve this?

Reply:
Added a TODO similar to the other places